src/data_cell/flatten_datacell.h

```diff
@@ -166,14 +166,13 @@ FlattenDataCell<QuantTmpl, IOTmpl>::BatchInsertVector(const float* vectors,
                                                       InnerIdType count,
                                                       InnerIdType* idx) {
     if (idx == nullptr) {
-        auto* codes = reinterpret_cast<uint8_t*>(
-            allocator_->Allocate(static_cast<uint64_t>(count) * static_cast<uint64_t>(code_size_)));
-        quantizer_->EncodeBatch(vectors, codes, count);
-        io_->Write(codes,
+        Vector<uint8_t> codes(static_cast<uint64_t>(count) * static_cast<uint64_t>(code_size_),
```

Comment:
This will repeatedly call memory allocation for memory allocation.

Reply:
done